File: REST/src/exceptions/MissingFieldException.php
<?php

namespace Exceptions;

class MissingFieldException extends RestException {

	private $missingField;


	/**
	 * Constructor
	 *
	 * @param	string     	$missingField		missing field name
	 */
	public function __construct(string $missingField) {
		parent::__construct(self::BAD_REQUEST, "MISSING_FIELD");

		$this->missingField = $missingField;
	}


	/**
	 * Get missing field name
	 *
	 * @return	string		missing field name
	 */
	public function getMissingField() : string {
		return $this->missingField;
	}


	/**
	 * Add exception specific data
	 *
	 * @param	array		$data		base level data
	 */
	protected function addErrorData(array &$data) {
		$data["field"] = $this->getMissingField();
	}

}

File: common/src/main/java/it/unishare/common/exceptions/MissingFieldException.java
package it.unishare.common.exceptions;

public class MissingFieldException extends Exception {

    private static final long serialVersionUID = -5085624064289030336L;
    private String missingField;


    /**
     * Constructor
     *
     * @param   missingField        missing field name
     */
    public MissingFieldException(String missingField) {
        this.missingField = missingField;
    }


    /**
     * Get missing field name
     *
     * @return  missing field name
     */
    public String getMissingField() {
        return missingField;
    }

}
